apps/fishing-map/features/datasets/upload/NewPointsDataset.tsx

```diff
@@ -134,6 +146,15 @@ function NewPointDataset({
     }
   }, [datasetMetadata, fileData, geojson, onConfirm, t, fileType])
 
+  const onDatasetPublicToggle = useCallback(() => {
+    setDatasetMetadata((meta) => ({
+      ...(meta as DatasetMetadata),
+      public: !!!meta?.public,
+    }))
+  }, [])
+
+  const isPublic = useMemo(() => !!datasetMetadata?.public, [datasetMetadata])
```

Comment:
In terms of performance, I don't think it is worth it to memoize this being a simple boolean